### PR TITLE
Perf/1 1

### PR DESCRIPTION
**Pull request by runspired**
## Todo

- [x] fix console usage (don't break on ie8,9)
- [x] minor cleanup (remove dead code)

## Summary

This PR refactors cooked+node to node+token

The token preserves `stats` `resume` and `stop` but will log a deprecation warning when used.

The new apis are

```
heimdall.stop(token);
heimdall.resume(token);
heimdall.statsForNode(token);
```

### CODE REVIEW

**Comment by runspired** [comment]
This is only failing in node 0.10 but for very good reason.  I'm using native `Map` without polyfill for mapping tokens to nodes at the moment.  We should decide what to put there in the long term.  I've had very good perf experiences using an EmptyObject wrapped in a Map like interface when keys are consistent. We could create one here that uses the ID as the key, and we could slyly add `_id` to the token that we are passing back to the user.

**Comment by runspired** [comment]
@hjdivad if you could confirm this works (obvs with deprecations flying around) with what you are doing, then I think we can merge and release.

This would be a minor release (deprecations only).

I'd also cut an almost immediate major release with the deprecated behavior stripped out, because once it is then I can merge the PR coming after this which strips out statements for `heimdall.stop();` `heimdall.start()` and `heimdall.resume();` when desired, which I think is necessary for us to merge this into ED in a way that we leave ED permanently instrumented without cost to end users.

**Comment by runspired** [comment]
Confirmed with @hjdivad that this does not break his current instrumentation project, we still need to check that this won't affect ember-cli.

At a minimum, we need to reconfigure the deprecation warnings to warn once so that we don't flood ember-cli users with thousands of warnings.

File: src/node.js [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
     this._id = heimdall.generateNextId();
     this.id = id;
 
     // TODO: strip for production builds for perfz
-    if (!(typeof this.id === 'object' && this.id !== null && typeof this.id.name === 'string')) {
-      throw new TypeError('HeimdallNode#id.name must be a string');
-     }
+    // if (!(typeof this.id === 'object' && this.id !== null && typeof this.id.name === 'string')) {
+    //  throw new TypeError('HeimdallNode#id.name must be a string');
+    // }
```
Comment (hjdivad) [inline src/node.js]: Let's just delete this.  If we want to add back in expensive assertions we'll need a .dev, .prod build

File: tests/heimdall.js [hunk 2]
```diff
@@ -80,25 +80,28 @@ describe('heimdall', function() {
...
-      cookieA.stop();
+      clock.tick(1, 10);
+      h1.stop(tokenA);
+
+      let statsA = h1.statsForNode(tokenA);
+      let statsB = h2.statsForNode(tokenB);
 
-      expect(cookieB._node.stats.time.self).to.equal(1e9 + 10);
+      expect(statsA.time.self).to.equal(2e9 + 20);
       // total A self time is time before B and time after B
```
Comment (hjdivad) [inline tests/heimdall.js]: need to move this comment up so it's stuck with the assertion for A

Comment (hjdivad) [inline tests/heimdall.js]: assertions swapped but comment remained put

File: src/node.js [hunk 3]
```diff
@@ -84,6 +81,7 @@ export default class HeimdallNode {
     this._children.push(node);
 
     node.parent = this;
+    node._restoreNode = node._restoreNode || this;
```
Comment (hjdivad) [inline src/node.js]: pretty sure this should just be `node._restoreNode = this`

**Pull request merged**